src/containers/MainDashboard/MainDashboard.tsx

```diff
@@ -62,39 +94,42 @@ const MainDashboard: React.FC = () => {
     setTitleInput(titleInput);
     setSelectionInput(selectionInput);
     const newRoadmap: any = await roadmap();
-    // client.writeData({ roadmaps: [...roadmaps, newRoadmap] })
-    setRoadmaps({ roadmaps: [...roadmaps, newRoadmap] });
+    const previousRoadmaps: any = client.cache.readQuery({ query: GET_LOCAL_ROADMAPS });
```

Comment:
Lines 97-100 expose this `MainDashboard` is too coupled to Apollo. I think you could use a `RoadmapsService` or a `DataService` that encapsulates all this logic in it, let's say a method called `addRoadmap`.